File: docs/get-images/request.js
// Get-Images RustOpus example

import { writeFile } from "fs/promises";

const AUTHCODE = "AAAAAAAA-0000-0000-0000-BBBBBBBBBBBB";
const OUTFILE = "get-images.xml";

const response = await fetch(`<url-to-rustopus>/get-images?authcode=${AUTHCODE}`);
const text = await response.text();

await writeFile(OUTFILE, text);

File: docs/api/get-bulk/request.py
'''
    Get-Bulk RustOpus example
'''

from urllib import request

# - - - - - -
# PARAMETERS
# - - - - - -
url: str = "<rustopus_url_here>"
get: str = "get-bulk"
authcode: str = "AAAAAAAA-0000-0000-0000-BBBBBBBBBBBB"
pid: int = 1234
outfile: str = "get-bulk.xml"

request_url: str = (f"{url}/{get}"
                    f"?authcode={authcode}"
                    f"pid={pid}")


with request.urlopen(url) as response:
    with open(file = outfile,
              mode = "wb") as f:
        f.write(response.read())
